src/main/java/net/schmizz/sshj/connection/channel/direct/SessionChannel.java

```diff
@@ -93,7 +98,7 @@ public Command exec(String command)
             throws ConnectionException, TransportException {
         checkReuse();
         log.debug("Will request to exec `{}`", command);
-        sendChannelRequest("exec", true, new Buffer.PlainBuffer().putString(command))
+        sendChannelRequest("exec", true, new Buffer.PlainBuffer().putString(command.getBytes(getRemoteCharset())))
```

Comment:
Or make a `putString(String s, Charset cs)` method, which makes the rest of the code at least cleaner and more concise.